tests/Agent/UnitTests/Core.UnitTest/AgentHealth/AgentHealthReporterTests.cs

```diff
@@ -45,19 +45,21 @@ public void TearDown()
             _configurationAutoResponder.Dispose();
         }
 
-        private static IConfiguration GetDefaultConfiguration()
+        private static IConfiguration GetDefaultConfiguration(bool enableLogging)
         {
             var configuration = Mock.Create<IConfiguration>();
             Mock.Arrange(() => configuration.LogEventCollectorEnabled).Returns(true);
             Mock.Arrange(() => configuration.LogDecoratorEnabled).Returns(true);
             Mock.Arrange(() => configuration.LogMetricsCollectorEnabled).Returns(true);
             Mock.Arrange(() => configuration.InfiniteTracingCompression).Returns(true);
+            Mock.Arrange(() => configuration.LoggingEnabled).Returns(enableLogging);
```

Comment:
You can probably avoid removing the SetUp attribute and needing to manually call SetUp from each test if you update this Mock to something more like the following if you make _enableLogging a field.
```suggestion
            Mock.Arrange(() => configuration.LoggingEnabled).Returns(() => _enableLogging);
```
Using that syntax will delay the evaluation of that field until the test needs to access configuration.LoggingEnabled.

Reply:
Oh that's clever. I'll do that.